test/dotcom_web/controllers/places_controller_test.exs

```diff
@@ -13,78 +19,89 @@ defmodule DotcomWeb.PlacesControllerTest do
 
   describe "autocomplete" do
     test "responds with predictions", %{conn: conn} do
-      input = "controller1"
-
-      autocomplete_fn = fn _, _ ->
-        {:ok, [%LocationService.Suggestion{address: "123 Sesame Street", highlighted_spans: []}]}
-      end
+      input = Faker.App.name()
+      hit_limit = Faker.random_between(3, 5)
+      suggestions = build_list(3, :suggestion)
 
-      conn = assign(conn, :autocomplete_fn, autocomplete_fn)
-      conn = get(conn, places_path(conn, :autocomplete, input, "3"))
+      expect(LocationService.Mock, :autocomplete, fn arg1, arg2 ->
+        assert arg1 == input
+        assert arg2 == hit_limit
```

Comment:
Using variable names like arg1 has as smell to it. Have you tried just pinning the values like `fn ^input, ^hit_limit ->`?